dependencyManagement/build.gradle.kts

```diff
@@ -0,0 +1,114 @@
+plugins {
+    `java-platform`
+}
+
+data class DependencySet(val group: String, val version: String, val modules: List<String>)
```

Comment:
To me, all this looks rather unfamiliar, not knowing kotlin. Do we really want to add another language to our buildscripts?

And is this `DependencySet` class, along with the boilerplate loop required to apply it, really better than inventing variable names for each dependency set?

Reply:
The plan is to remove Groovy since lack of completion / compile safety makes it quite tedious for me to maintain the scripts. We already have kotlin in opentelemetry-kotlin, after a migration, we would have one less programming language, Groovy, though.

> And is this DependencySet class, along with the boilerplate loop required to apply it, really better than inventing variable names for each dependency set?

I think it's an implementation detail of this script so that the downstream projects just have to use maven coordinates for defining dependencies. It's more decoupling of the base and downstream projects than we have right now.